process/dns.go

```diff
@@ -0,0 +1,188 @@
+package process
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/brexhq/substation/condition"
+	"github.com/brexhq/substation/config"
+)
+
+var dnsResolver net.Resolver
```

Comment:
refactor: the stdlib `net.Resolver` won't cache DNS responses, sometimes in a pipelines handling of capsules the same IP might be seen more than one time, do you think it makes sense to introduce a cache in this package to save the forward and reverse lookups that have already been done?

Reply:
Great point on considering apps not in-repo, a short lived map would work fine and be effective due until the Lambda/local run restarts but we'll need to consider longer lived scenarios like a K8s environment or similar.

Since posting I realized for DNS specifically this could be less of a concern due to OS level caching of DNS responses, but maybe an implementation like Logstash's caching could be useful for future processors!